src/installer/installer.go

```diff
@@ -320,42 +315,43 @@ func (i *installer) waitForController() error {
 }
 
 // wait for minimum master nodes to be in ready status
-func (i *installer) waitForMasterNodes(ctx context.Context, minMasterNodes int, kc k8s_client.K8SClient) error {
-	nodesTimeout := time.Duration(i.Config.InstallationTimeout) * time.Minute
-
-	if nodesTimeout < 0 {
-		return fmt.Errorf("Nodes installation timeout %d multiplication by minutes caused an integer overflow", i.Config.InstallationTimeout)
-	}
+func (i *installer) waitForMasterNodes(ctx context.Context, minMasterNodes int, kc k8s_client.K8SClient) {
 
 	var readyMasters []string
 	var inventoryHostsMap map[string]inventory_client.HostData
-	i.log.Infof("Waiting up to %v for %d master nodes", nodesTimeout, minMasterNodes)
-	apiContext, cancel := context.WithTimeout(ctx, nodesTimeout)
-	defer cancel()
-
-	wait.Until(func() {
+	i.log.Infof("Waiting for %d master nodes", minMasterNodes)
+	sufficientMasterNodes := func() bool {
 		var err error
 		inventoryHostsMap, err = i.getInventoryHostsMap(inventoryHostsMap)
 		if err != nil {
-			return
+			return false
 		}
 		nodes, err := kc.ListMasterNodes()
 		if err != nil {
 			i.log.Warnf("Still waiting for master nodes: %v", err)
-			return
+			return false
 		}
 		i.updateReadyMasters(nodes, &readyMasters, inventoryHostsMap)
 		i.log.Infof("Found %d ready master nodes", len(readyMasters))
 		if len(readyMasters) >= minMasterNodes {
 			i.log.Infof("Waiting for master nodes - Done")
-			cancel()
+			return true
+		}
+		return false
+	}
+
+	for {
```

Comment:
maybe lets add waitForPredicateWithContext? i will need to another function for sure?

Reply:
In another PR